plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestingSqlServer.java

```diff
@@ -17,25 +17,43 @@
 import org.testcontainers.utility.DockerImageName;
 
 import java.sql.Connection;
+import java.sql.SQLException;
 import java.sql.Statement;
 
+import static java.util.Objects.requireNonNull;
+
 public final class TestingSqlServer
         extends MSSQLServerContainer<TestingSqlServer>
 {
+    private String catalogName = "master";
+
     public TestingSqlServer()
     {
         super(DockerImageName.parse("microsoft/mssql-server-linux:2017-CU13").asCompatibleSubstituteFor("mcr.microsoft.com/mssql/server:2017-CU12"));
         addEnv("ACCEPT_EULA", "yes");
     }
 
+    public void setCurrentCatalog(String catalogName)
+    {
+        this.catalogName = requireNonNull(catalogName, "catalogName is null");
+    }
+
     public void execute(String sql)
     {
-        try (Connection connection = createConnection("");
+        try (Connection connection = connect();
                 Statement statement = connection.createStatement()) {
             statement.execute(sql);
         }
         catch (Exception e) {
             throw new RuntimeException("Failed to execute statement: " + sql, e);
         }
     }
+
+    public Connection connect()
+            throws SQLException
+    {
+        Connection connection = createConnection("");
```

Comment:
the catalogName should be weaved in within `constructUrlForConnection`, so it's used when user calls `createConnection("")` or any other API method.

Also, we should consider replacing `estingSqlServer        extends MSSQLServerContainer` with composition to keep the API under control (certainly out of scope for this PR though)